syntax = "proto2";
import "aibsmw_messages.proto";

message config_loaded {
    required message_header header = 1;
    required string version = 2;
}

message capture_image {
    required message_header header = 1;
    required string image_type = 2;
}

message image_captured {
    required message_header header = 1;
    required string image_type = 2;
    required string image_path = 3;
    required float x_stage = 4;
    required float y_stage = 5;
    required float z_stage = 6;
    required string timestamp = 7;
    required float x_size = 8;
    required float y_size = 9;
    required string working_units = 10;
}

message get_z_depth {
    required message_header header = 1;
}

message z_depth {
   required message_header header = 1;
   required float z_stage = 2;
   required string working_units = 3;
}

message set_link_btn_state {
    required message_header header = 1;
    required bool checked = 2;
}

message set_surface_btn {
    required message_header header = 1;
    required bool enabled = 2;
}

message clear_tile_images {
    required message_header header = 1;
}

message load_tiled_image {
    required message_header header = 1;
    required string img_path = 2;
}